test/integration/kubernetes_deploy_test.rb

```diff
@@ -890,4 +890,50 @@ def test_cronjobs_can_be_deployed
     cronjobs = FixtureSetAssertions::CronJobs.new(@namespace)
     cronjobs.assert_cronjob_present("my-cronjob")
   end
+
+  def test_resource_watcher_reports_failed_after_timeout
+    result = deploy_fixtures(
+      "invalid",
+      subset: ["bad_probe.yml", "cannot_run.yml", "missing_volumes.yml"],
+      max_watch_seconds: 15
+    ) do |f|
+      deployment = f["bad_probe.yml"]["Deployment"].first
+      deployment["metadata"]["annotations"]["kubernetes-deploy.shopify.io/timeout-override"] = '5s'
+      f["cannot_run.yml"]["Deployment"].first["spec"]["replicas"] = 1
+      f["bad_probe.yml"]["ConfigMap"] = {
```

Comment:
unimportant, but I'd suggest including this configmap in `bad_probe.yml` since both tests that use that fixture are manually adding this configmap (I probably should have put it in the fixture in the first place).